core/lib/types/src/api/mod.rs

```diff
@@ -736,3 +737,38 @@ pub struct Proof {
     pub address: Address,
     pub storage_proof: Vec<StorageProof>,
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OptimisticTransactionResult {
+    pub transaction_hash: H256,
+    pub storage_logs: Vec<ApiStorageLogQuery>,
+    pub events: Vec<ApiVmEvent>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ApiVmEvent {
+    pub l1_batch_number: L1BatchNumber,
+    pub tx_batch_index: u32,
+    pub address: Address,
+    pub indexed_topics: Vec<H256>,
+    pub value: String,
```

Comment:
Are you sure we need a new `struct` for this? We already return logs in API, please check out `Log` struct. 

I understand it has a larger set of fields, but IMO consistency is very important here. These logs will be processed by Lens the same way as they process them in transaction receipt, so their code would need to understand both notations (e.g. `indexed_topics` vs `topics`, `transaction_index` vs `tx_batch_index` etc. 

Additionally, I think `struct Log` follows Ethereum specification.

Reply:
makes sense - will use `Log` instead. We will have to use a custom struct for vm events though, yes?

Reply:
Fixed, as discussed